### PR TITLE
Added `TreeNode.is_bifurcating`

### PR DESCRIPTION
**Pull request by qiyunzhu**
Please complete the following checklist:

* [x] I have read the [contribution guidelines](https://scikit.bio/contribute.html).

* [x] I have documented all public-facing changes in the [changelog](https://github.com/scikit-bio/scikit-bio/blob/main/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/scikit-bio/scikit-bio/blob/main/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied.

  - **It is your responsibility to disclose** code, documentation, or other content derived from external source(s). If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] This pull request does not include code, documentation, or other content derived from external source(s).

Note: [This document](https://scikit.bio/devdoc/review.html) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

File: skbio/tree/_tree.py [hunk 1]
```diff
@@ -3116,6 +3119,54 @@ def assign_supports(self):
...
+        if strict:
+            for node in self.non_tips(include_self=True):
+                if len(node.children) != 2:
+                    return False
+            return True
+        else:
+            for node in self.traverse(include_self=True):
+                if len(node.children) > 2:
+                    return False
+            return True
```
Comment (wasade) [inline skbio/tree/_tree.py:3168]: A redundant return can be removed with:

```suggestion
        else:
            for node in self.traverse(include_self=True):
                if len(node.children) > 2:
                    return False
        return True
```

A further reduction in replication could be done with the below, though for something like this I think it's really just a stylistic item. 

```python
import operator

if strict:
    test = operator.ne
else:
    test = operator.gt

for node in self.non_tips(include_self=True):
    if test(len(node.children), 2):
        return False

return True
```

**Review by wasade — APPROVED** [review]
Just a minor comment. And, I think it would be nice to note in the ChangeLog

File: skbio/tree/_tree.py (hunk 1 above)
Comment (qiyunzhu) [reply skbio/tree/_tree.py:3168]: @wasade Interesting! Your suggestion is very elegant. I think that `test` as a function call may be slower than a straight operator, but perhaps Python has optimization for this. Let me benchmark it.

Comment (wasade) [reply skbio/tree/_tree.py]: Possibly though this isn't a performance critical method unless there is a reasonable need to run `is_bifurcating` a very large number of times on the same tree?

**Comment by qiyunzhu** [comment]
Hi @wasade I have benchmarked, and realized that the true bottleneck is `non_tips`, whereas the operator functions you suggested has minimum impact on the performance. Therefore, I have updated the PR to adopt your suggestion, with slight modifications. Here are the benchmarks:

Version 1:
```python
if strict:
    for node in self.traverse(include_self=True):
        if node.children and len(node.children) != 2:
            return False
else:
    for node in self.traverse(include_self=True):
        if len(node.children) > 2:
            return False
return True
```

Version 2:
```python
test = ne if strict else gt
for node in self.non_tips(include_self=True):
    if test(len(node.children), 2):
        return False
return True
```

Version 3:
```python
test = ne if strict else gt
for node in self.traverse(include_self=True):
    if (children := node.children) and test(len(children), 2):
        return False
return True
```

Benchmarks on the 10.5k-tip tree:

```
v1: 1.83 ms ± 44.2 μs
v2: 3.66 ms ± 96.6 μs
v3: 1.88 ms ± 7.13 μs
v1, strict: 1.81 ms ± 7.42 μs
v2, strict: 3.61 ms ± 72.8 μs
v3, strict: 1.95 ms ± 42.2 μs
```

**Pull request merged**